icu4c/source/test/intltest/caltest.cpp

```diff
@@ -5929,6 +5930,25 @@ void CalendarTest::TestAddOverflow() {
         }
     }
 }
+
+void CalendarTest::Test22750Roll() {
+    UErrorCode status = U_ZERO_ERROR;
+    Locale l(Locale::getRoot());
+    std::unique_ptr<icu::StringEnumeration> enumeration(
+        Calendar::getKeywordValuesForLocale("calendar", l, false, status));
+    // Test every calendar
+    for (const char* name = enumeration->next(nullptr, status);
+        U_SUCCESS(status) && name != nullptr;
+        name = enumeration->next(nullptr, status)) {
```

Comment:
indentation so that the condition does not blend into the statements block
```suggestion
            U_SUCCESS(status) && name != nullptr;
            name = enumeration->next(nullptr, status)) {
```